[Mobile Payments] Use currency formatter in Yosemite

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of: #5108 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
With this PR we use `WooFoundation.CurrencyFormatter` in `Yosemite` when amount formatting is necessary (receipts), making it consistent with the rest of the codebase. Furthermore, since the logic of `CurrencyFormatter` is more sophisticated, we ensure that the receipt amounts formatting will handle all cases.
Since the receipt formatting logic was already covered by unit tests, it wasn't necessary to add any here. As they pass successfully, we can be sure that adding `CurrencyFormatter` didn't cause any format changes.
### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

#### Create an order

1. Go to Orders tab
2. Tap on + to create an order
3. After the order is created, collect the payment
4. When the payment collection process finishes, tap on print the receipt
5. On the Print Options receipt preview, make sure that the amount formatting is correct


![Simulator Screen Shot - iPhone 13 - 2022-06-07 at 18 14 49](https://user-images.githubusercontent.com/1864060/172430830-04979918-ee2f-4103-bca2-47a05a0e8fa7.png)

#### Completed Order

1. Go to Orders tab
2. Go to a completed Order
3. Tap on See receipt
4. On the receipt detail view, the amounts should be formatted properly

![Simulator Screen Shot - iPhone 13 - 2022-06-07 at 18 13 44](https://user-images.githubusercontent.com/1864060/172430922-1fe1a427-5ba0-4369-b716-c5862c21085e.png)

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

toupper committed:
Use CurrencyFormatter in Yosemite

toupper committed:
Add TODO that was wrongly removed

koke reviewed (commented):
Since we're introducing `CurrencyFormatter`, maybe we can also use it to format the amounts with their currency symbol instead of appending the currency code afterwards.

Also noting that the argument labels in the `CurrencyFormatter` are super confusing (`using`, `at`, `with`, `in`, `including`,...) 😬 . That's not really related to this PR, but if we're working on extracting it into a library, it might be a good opportunity to polish the API a little bit.

toupper committed:
Replace currency code with currency symbol now that we have CurrencyFormatter available.

toupper committed:
Adapt unit tests to new currency formatter logic.

toupper committed:
Refactor parameter naming for convertToDecimal function

toupper committed:
Improve CurrencyFormatter and settings API

toupper commented:
Thanks for your remarks @koke
I updated the PR to use the currency symbol in the right place, as well as refactored the `CurrencyFormatter` and `CurrencySettings` API to make it more readable. Likewise, I renamed some elements to align them with Apple syntax (e.g`groupingSeparator`, `fractionDigits`... ) I left the original values in `WooCommerce` and `Yosemite` target though, to make them consistent with the Woo Backend syntax.

koke commented:
Receipts look fine, and the API is much cleaner now ✨
